kustomize/components/alloydb/README.md

```diff
@@ -0,0 +1,152 @@
+# Integrate Online Boutique with AlloyDB
+
+By default the `cartservice` stores its data in an in-cluster Redis database. 
+Using a fully managed database service outside your GKE cluster (such as [AlloyDB](https://cloud.google.com/alloydb)) could bring more resiliency and more security.
+
+Note that because of AlloyDB's current connectivity, you'll need to run all this from a VM with
+VPC access to the network you want to use for everything (out of the box this should just use the
+default network). The Cloud Shell doesn't work because of transitive VPC peering not working.
+
+## Provision an AlloyDB database and the supporting infrastructure
+
+# Environmental variables needed for setup. These should be set in a .bashrc or similar as some of the variables are used in the application itself. Default values are supplied in this readme, but any of them can be changed.
```

Comment:
Does this mean that you need to run these steps (and set these ENV variables) on the specific VM where `cartservice` will run?

Reply:
It's a combination. They need to be set for various parts of the set up here (where all the set up calls are made from, in our case a GCE instance since that's the only place we can access the AlloyDB instance), and they ALSO need to be set in the kustomization YAML. The sed calls below handle updating that file for the environment of the container, but they DO need to be set where the sed is being called from so that the vars can be picked up in the substitution.